packages/fast-components/src/anchored-region/anchored-region.ts

```diff
@@ -0,0 +1,978 @@
+        this.anchorLeft = anchorEntry.boundingClientRect.left;
+        this.anchorHeight = anchorEntry.boundingClientRect.height;
+        this.anchorWidth = anchorEntry.boundingClientRect.width;
+    };
+
+    /**
+     *  Update data based on positioner collisions
+     */
+    private handlePositionerCollision = (
+        positionerEntry: IntersectionObserverEntry,
+        shouldDeriveAnchorPosition: boolean
+    ): void => {
+        this.viewportRect = positionerEntry.rootBounds;
+        const positionerRect: ClientRect | DOMRect = positionerEntry.boundingClientRect;
+        this.positionerDimension = {
+            height: positionerRect.height,
+            width: positionerRect.width,
+        };
+
+        if (shouldDeriveAnchorPosition) {
+            switch (this.currentVerticalPosition) {
+                case AnchoredRegionVerticalPositionLabel.top:
+                    this.anchorTop = positionerRect.bottom;
+                    this.anchorBottom = this.anchorTop + this.anchorHeight;
+                    break;
+
+                case AnchoredRegionVerticalPositionLabel.insetTop:
+                    this.anchorBottom = positionerRect.bottom;
+                    this.anchorTop = this.anchorBottom - this.anchorHeight;
+                    break;
+
+                case AnchoredRegionVerticalPositionLabel.insetBottom:
+                    this.anchorTop = positionerRect.top;
+                    this.anchorBottom = this.anchorTop + this.anchorHeight;
+                    break;
+
+                case AnchoredRegionVerticalPositionLabel.bottom:
+                    this.anchorBottom = positionerRect.top;
+                    this.anchorTop = this.anchorBottom - this.anchorHeight;
+                    break;
+            }
+
+            switch (this.currentHorizontalPosition) {
+                case AnchoredRegionHorizontalPositionLabel.left:
+                    this.anchorLeft = positionerRect.right;
+                    this.anchorRight = this.anchorLeft + this.anchorWidth;
+                    break;
+
+                case AnchoredRegionHorizontalPositionLabel.insetLeft:
+                    this.anchorRight = positionerRect.right;
+                    this.anchorLeft = this.anchorRight - this.anchorWidth;
+                    break;
+
+                case AnchoredRegionHorizontalPositionLabel.insetRight:
+                    this.anchorLeft = positionerRect.left;
+                    this.anchorRight = this.anchorLeft + this.anchorWidth;
+                    break;
+
+                case AnchoredRegionHorizontalPositionLabel.right:
+                    this.anchorRight = positionerRect.left;
+                    this.anchorLeft = this.anchorRight - this.anchorWidth;
+                    break;
+            }
+        }
+    };
+
+    /**
+     *  Handle region resize events
+     */
+    private handleRegionResize = (): void => {
+        const regionHeight: number = this.region !== null ? this.region.clientHeight : 0;
```

Comment:
Doesn't seem like you need to do this work unless you enter the if cases - consider moving these inside those statements.

Reply:
done